cmd/gazctl/gazctlcmd/shards_prune.go

```diff
@@ -123,6 +127,16 @@ func (cmd *cmdShardsPrune) Execute([]string) error {
 	return nil
 }
 
+func overlapsAnySegment(segments []recoverylog.Segment, fragment pb.Fragment) bool {
+	for _, seg := range segments {
+		if (seg.FirstOffset < fragment.End || fragment.End == 0) &&
```

Comment:
> I presume you did the same.

Yeah, and also updated the tests to use this function so that we're exercising it. I waffled a bit on whether to add unit tests for just this function, and would be happy to do so if you think it'd help